src/p2p/net_peerlist.h

```diff
@@ -214,6 +215,16 @@ namespace nodetool
     }
   }
   //--------------------------------------------------------------------------------------------------
+  inline
+  peerlist_entry peerlist_manager::get_nth_latest_peer(peers_indexed& peerlist, const size_t n)
+  {
+    // Is not thread-safe nor does it check bounds. Do this before calling. Indexing starts at 0.
+    peers_indexed::index<by_time>::type& by_time_index = peerlist.get<by_time>();
+    auto by_time_it = --by_time_index.end();
+    std::advance(by_time_it, -((long long) n));
```

Comment:
```C++
std::advance(by_time_it, -(static_cast<long long>(n)));
```
How did it behave without the cast, BTW?

Reply:
I didn't try without a cast because n is type `size_t` and `size_t` is unsigned, so I'm pretty sure `-n` would be undefined. Even if it isn't undefined, I'm sure some compiler would throw out a warning regarding negating an unsigned type. I will change it to a `static_cast<>` though!

Reply:
I understand. OK this makes sense.